### ISSUE TITLE
`isinstance` inference seems to only work if done in the `namedexpr_test` 

### ISSUE THREAD

**Issue by hackaugusto** [python/mypy#8390]
Note: if you are reporting a wrong signature of a function or a class in
the standard library, then the typeshed tracker is better suited
for this report: https://github.com/python/typeshed/issues

Please provide more information to help us understand the issue:

* Are you reporting a bug, or opening a feature request?

I believe this is a bug.

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.

```python
class Base:
    pass

class Sub(Base):
    def __init__(self):
        self.a = 1

def f(v: Base):
    is_working = isinstance(v, Sub)
    if is_working:
        print(v.a)
```

* What is the actual behavior/output?
```
t.py:13: error: "Base" has no attribute "a"
Found 1 error in 1 file (checked 1 source file)
```
* What is the behavior/output you expect?

I expect the `isinstance` check should narrow the type if the test is not directly in the `namedexpr_test` 

* What are the versions of mypy and Python you are using?
  Do you see the same issue after installing mypy from Git master?

```
mypy 0.761
mypy 0.720
mypy 0.770+dev.1104a9f1cf44a0f8469b4ce82b9fc5e75539195f
Python 3.8.1
```

* What are the mypy flags you are using? (For example --strict-optional)

None

* If mypy crashed with a traceback, please paste
  the full traceback below.

**Comment by ilevkivskyi** [comment]
It is unlikely this will be supported in foreseeable future, sorry.
